### DISCUSSION TITLE
Dreaded SSL Certificate issue - no such file or directory

### DISCUSSION THREAD

**Discussion by nyfon** [postalserver/postal#2626] [Help with using Postal] [answered]
Running postal from the command line under root.
The cert and key files are copied to /opt/postal/config and chmod-ed to 777.
Still getting  Errno::ENOENT: No such file or directory @ rb_sysopen - /opt/postal/config/smtp.crt
Tried default names - same issue. 
Could you please advise?

Postal 2.1.4 
AlmaLinux 9.2
selinux disabled

**Comment by willpower232** [comment]
Did you use a volume in the docker-compose.yml file to bring the certificates in to the container?

**Reply by nyfon** [reply]
I think I did. Initialized with the standard docker-compose.yml:


version: "3.9"
services:
  web:
    image: ghcr.io/postalserver/postal:2.1.4
    command: postal web-server
    network_mode: host
    volumes:
      - /opt/postal/config:/config

  smtp:
    image: ghcr.io/postalserver/postal:2.1.4
    command: postal smtp-server
    network_mode: host
    cap_add:
      - NET_BIND_SERVICE
    volumes:
      - /opt/postal/config:/config

  worker:
    image: ghcr.io/postalserver/postal:2.1.4
    command: postal worker
    network_mode: host
    volumes:
      - /opt/postal/config:/config

  cron:
    image: ghcr.io/postalserver/postal:2.1.4
    command: postal cron
    network_mode: host
    volumes:
      - /opt/postal/config:/config

  requeuer:
    image: ghcr.io/postalserver/postal:2.1.4
    command: postal requeuer
    network_mode: host
    volumes:
      - /opt/postal/config:/config

  runner:
    profiles: ["tools"]
    image: ghcr.io/postalserver/postal:2.1.4
    command: postal
    network_mode: host
    volumes:
      - /opt/postal/config:/config

**Reply by willpower232** [reply]
> /opt/postal/config:/config

this means that the contents of /opt/postal/config on the host are in /config in the container so the path in the config file needs to start /config so hopefully you don't need to mess around with `docker cp`

**Reply by nyfon** [reply]
That is a weird thing - might be the centos/alma issue. Files are in /opt/postal/config but when I run postal console, /config is empty. If I do the cp they are appear there.  Maybe cert and key be placed into /opt/postal/config before bootstrap? But the signing.key and postal.yml are not in container's /config either...

**Comment by nyfon** [comment, marked as answer]
I kind of figured whats going on. So basically I had to manually copy certificate and key file to the smtp container... Now it works.
Is there a way to automate it? I'm not an expert in docker and ruby...

If anyone struggles with default installation - here is what I did to resolve this:
1. Start postal - "postal start".
2. List running containers- "docker ps"
3. Choose the container id with the postal-smtp-1 name.
4. Copy each cert and key files to it: docker cp /path/to/cert/cert.crt containerid:/config/ && docker cp /path/to/key/cert.key containerid:/config/

**Comment by sinnbeck** [comment]
I made a small script that I run in a cronjob. Replace [yourdomain] with the actual domain of the server. I use the certificate from caddy btw (just changed the type it generates)

```
cp /opt/postal/caddy-data/caddy/certificates/acme-v02.api.letsencrypt.org-directory/[yourdomain]/[yourdomain].crt /opt/postal/config/smtp.crt
cp /opt/postal/caddy-data/caddy/certificates/acme-v02.api.letsencrypt.org-directory/[yourdomain]/[yourdomain].key /opt/postal/config/smtp.key
chmod o+r /opt/postal/config/smtp.*
postal stop && postal start
